### PR TITLE
TMA-488 Rollout brick: option to make full sync of processes (including delete)

### PR DESCRIPTION
**Pull request by vhtien**
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog
- [ ] Build passing

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/759/ : SUCCESS in 29s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/714/ : SUCCESS in 45m 53s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/484/ : FAILURE in 44s

File: lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb [hunk 1]
```diff
@@ -81,13 +81,42 @@ def call(params)
             end
           end
 
-          params.synchronize.peach do |info|
+          delete_extra_process_schedule = GoodData::Helpers.to_boolean(params.delete_extra_process_schedule)
+
+          params.synchronize.peach do |info| # rubocop:disable Metrics/BlockLength
```
Comment (panjan) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: rubocop

File: lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb [hunk 2]
```diff
@@ -81,13 +81,42 @@ def call(params)
             end
           end
```
Comment (panjan) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: Can you write some specs for this please?

Comment (panjan) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: Do we have documentation for the new parameter?

File: lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb [hunk 3]
```diff
@@ -81,13 +81,42 @@ def call(params)
...
+              from_project_process_names = from_project_processes.map(&:name)
+              from_project_schedule_names = from_project.schedules.map { |schedule| [schedule.name, from_project_process_id_names[schedule.process_id]] }
+            end
+
             to_projects = info.to
             to_projects.peach do |entry|
               pid = entry[:pid]
               to_project = client.projects(pid) || fail("Invalid 'to' project specified - '#{pid}'")
+
+              if delete_extra_process_schedule
```
Comment (panjan) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: Would it be better to name the parameter `delete_extra_processes_and_schedules` or `delete_extra_etl`?

File: lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb (hunk 2 above)
Comment (panjan) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: Please update the changelog.

File: lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb (hunk 3 above)
Comment (vhtien) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: that parameter name is approved by ZD and Petr Valkoun

File: lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb (hunk 2 above)
Comment (vhtien) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: the document will be updated in confluence

Comment (vhtien) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: The test will be in gerrit

File: lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb (hunk 1 above)
Comment (vhtien) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: please ignore this rubocop because I have to balance with the performance

File: lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb (hunk 2 above)
Comment (vhtien) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: this is LCM2, not related to SDK so no need to update changelog

**Comment by vhtien** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/770/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/726/ : SUCCESS in 45m 05s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/496/ : FAILURE in 24s

File: lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb (hunk 2 above)
Comment (panjan) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: The integration is going to test the happy path. It would be nice to write unit tests for the corner cases.

File: lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb (hunk 1 above)
Comment (panjan) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: I believe that code performance is unrelated to coding style. The BlockLength rule makes a perfect sense here as the method is getting very lengthy. One way to go about this would be extracting the bodies of the `if`s into private methods.

File: lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb (hunk 2 above)
Comment (vhtien) [inline lib/gooddata/lcm/actions/synchronize_etls_in_segment.rb]: I'm sorry. I don't like the way mocking test.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/773/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/729/ : SUCCESS in 46m 07s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/499/ : UNSTABLE in 19m 56s
